core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -55,6 +58,24 @@ private[spark] class IndexShuffleBlockResolver(
 
   def getDataFile(shuffleId: Int, mapId: Long): File = getDataFile(shuffleId, mapId, None)
 
+  /**
+   * Get the shuffle files that are stored locally. Used for block migrations.
+   */
+  override def getStoredShuffles(): Set[ShuffleBlockInfo] = {
+    // Matches ShuffleIndexBlockId name
+    val pattern = "shuffle_(\\d+)_(\\d+)_.+\\.index".r
```

Comment:
feels like this should live in BlockId related to ShuffleIndexBlockId

Reply:
Sounds reasonable. I think there is a similar regex around the codebase somewhere, I'll see if I can unify them (or that could just be excess coffee).

Reply:
I did some poking and it seems like we can drop this regex and instead do some pattern matching off of getAllBlocks from the disk manager. Thanks for pointing out this was weird.